deps/wazuh_testing/wazuh_testing/logcollector.py

```diff
@@ -357,6 +389,90 @@ def add_log_data(log_path, log_line_message, size_kib=1024, line_start=1, print_
     return 0
 
 
+def get_data_sending_stats(log_path, socket_name, state_interval):
+    """Returns the statistics of a log monitored by logcollector.
+
+    For this purpose, it parses the "wazuh-logcollector.state" file and retrieves the data.
+    See:
+    https://documentation-dev.wazuh.com/current/user-manual/reference/statistics-files/wazuh-logcollector-state.html
+
+    Args:
+        log_path (str): Path of the log from which the statistics are to be obtained.
+        socket_name (str): Target socket name.
+        state_interval (int): Statistics generation interval, in seconds ("logcollector.state_interval").
+
+    Returns:
+        dict: Dictionary with the statistics.
+    """
+    wait_statistics_file()
+
+    if not path.isfile(LOGCOLLECTOR_STATISTICS_FILE):
+        raise TimeoutError
+
+    with open(LOGCOLLECTOR_STATISTICS_FILE, 'r') as json_file:
+        data = load(json_file)
+        global_files = data['global']['files']
+        global_start_date = datetime.strptime(data['global']['start'], '%Y-%m-%d %H:%M:%S')
+        global_end_date = datetime.strptime(data['global']['end'], '%Y-%m-%d %H:%M:%S')
+        interval_files = data['interval']['files']
+        interval_start_date = datetime.strptime(data['interval']['start'], '%Y-%m-%d %H:%M:%S')
+        interval_end_date = datetime.strptime(data['interval']['end'], '%Y-%m-%d %H:%M:%S')
+        stats = {'global_events': 0, 'global_drops': 0,
+                 'global_start_date': global_start_date, 'global_end_date': global_end_date,
+                 'interval_events': 0, 'interval_drops': 0,
+                 'interval_start_date': interval_start_date, 'interval_end_date': interval_end_date}
+        # Global statistics
+        for g_file in global_files:
+            if g_file['location'] == log_path:
+                stats['global_events'] = g_file['events']
+                targets = g_file['targets']
+                for t in targets:
+                    if t['name'] == socket_name:
+                        stats['global_drops'] = t['drops']
+        # Interval statistics
+        for i_file in interval_files:
+            if i_file['location'] == log_path:
+                stats['interval_events'] = i_file['events']
+                targets = i_file['targets']
+                for t in targets:
+                    if t['name'] == socket_name:
+                        stats['interval_drops'] = t['drops']
```

Comment:
Same

Reply:
Done in: https://github.com/wazuh/wazuh-qa/pull/1338/commits/058ef4e5b590a99da16710e8a6780e64516906fa